actionpack/lib/action_dispatch/http/response.rb

```diff
@@ -5,6 +5,14 @@
 require "action_dispatch/http/cache"
 require "monitor"
 
+begin
+  # For `Rack::Headers` (Rack 3+):
+  require "rack/headers"
+rescue LoadError
+  # For `Rack::Utils::HeaderHash`:
+  require "rack/utils"
```

Comment:
If acceptable by the style guide, I would move this to the `Headers` definition, so you don't need two conditionals.

Reply:
Implemented.